integration/src/test/java/org/mule/test/streaming/PayloadStatisticsTestCase.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.streaming;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.apache.commons.io.FileUtils.writeStringToFile;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mule.test.allure.AllureConstants.StreamingFeature.STREAMING;
+import static org.mule.test.allure.AllureConstants.StreamingFeature.StreamingStory.STATISTICS;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.core.api.construct.Flow;
+import org.mule.runtime.core.api.management.stats.PayloadStatistics;
+import org.mule.tck.junit4.rule.SystemProperty;
+import org.mule.tck.util.StatisticsEnabled;
+import org.mule.test.AbstractIntegrationTestCase;
+import org.mule.tests.api.TestQueueManager;
+
+import java.io.File;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+
+@Feature(STREAMING)
+@Story(STATISTICS)
+public class PayloadStatisticsTestCase extends AbstractIntegrationTestCase {
+
+  @ClassRule
+  public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Rule
+  public SystemProperty workingDirSysProp = new SystemProperty("workingDir", temporaryFolder.getRoot().getPath());
+
+  @Inject
+  private TestQueueManager queueManager;
+
+  @Inject
+  @Named("listOfMessagesSource")
+  public Flow listOfMessagesSource;
+
+  public StatisticsEnabled withStatistics;
+
+  @Override
+  protected String getConfigFile() {
+    return "org/mule/streaming/payload-statistics-config.xml";
+  }
+
+  @Before
+  public void before() throws Throwable {
+    withStatistics = new StatisticsEnabled(() -> muleContext);
```

Comment:
Maybe you can use the system property instead of the rule now after it is merged. Though we can merge this as is.